EpikCord/ext/tasks.py

```diff
@@ -0,0 +1,83 @@
+import datetime
+from typing import Callable, Optional
+import asyncio
+from ..exceptions import TaskFailedError
+
+_EPOCH = datetime.datetime(1970, 1, 1)
+__all__ = (
+    "TimeParser",
+    "Tasks",
+)
+
+
+
+
+
+def total_seconds(
+    day: int = 0,
+    hour: int = 0,
+    minute: int = 0,
+    seconds: float = 0,
+) -> float:
+    hour += day * 24
+    minute += hour * 60
+    seconds += minute * 60
+    return seconds
+
+
+class Tasks:
+    def __init__(self, client) -> None:
+        self.client = client
+
+    def add_task(
+        self,
+        task: Callable[..., None],
+        interval: Optional[int] = None,
```

Comment:
make everything optional a kwarg, and maybe add a decorator for ease.

Reply:
But then you said to make all vars explicit so that it doesnt get passed to the actual function